tests/benchq/compilation/test_rbs_with_pauli_tracking.py

```diff
@@ -0,0 +1,574 @@
+#!/usr/bin/env python
+import os
+import random
+
+import numpy as np
+import pytest
+import qiskit
+from orquestra.quantum.circuits import CNOT, CZ, RZ, Circuit, H, I, S, T, X, Y, Z
+from qiskit import Aer, ClassicalRegister, QuantumCircuit, QuantumRegister
+from qiskit.transpiler.passes import RemoveBarriers
+
+from benchq.compilation.graph_states import jl
+from benchq.conversions import export_circuit
+from benchq.visualization_tools.plot_graph_state import plot_graph_state
+
+np.random.seed(0)
+
+SKIP_SLOW = pytest.mark.skipif(
+    os.getenv("SLOW_BENCHMARKS") is None,
+    reason="Slow benchmarks can only run if SLOW_BENCHMARKS env variable is defined",
+)
+
+
+def verify_random_circuit(n_qubits, depth, hyperparams, optimization, verbose):
+    # test that a single random circuit works for |0> initialization
+
+    circuit = generate_random_circuit(n_qubits, depth)
+
+    check_correctness_for_single_init(
+        circuit,
+        # Initialize in all possible single qubit stabiilzer states
+        Circuit([H(0), H(1), H(2), H(3), H(4), H(5), S(3), S(4), S(5)]),
+        hyperparams,
+        show_circuit=True,
+        throw_error_on_incorrect_result=True,
+        optimization=optimization,
+        verbose=verbose,
+    )
+
+
+def generate_random_circuit(n_qubits, depth):
```

Comment:
This is the function I was thinking of when we were talking about making the tests more deterministic. One option could be to replace the calls to random with calls to np.random since we are already setting a seed for it.

Reply:
Seeded!